lib/stitch_plan/stitch_plan.py

```diff
@@ -39,6 +39,9 @@ def stitch_groups_to_stitch_plan(stitch_groups, collapse_len=None, min_stitch_le
 
         if color_block.color != stitch_group.color:
             if len(color_block) == 0:
+                # end the previous block with a color change
```

Comment:
This fix looks good to me (I think).  We should probably move the comment about "we just processed a stop" up to before this line.